File: ios-native/DMVCalifornia/Core/Models/Flashcard.swift
import Foundation

// MARK: - Flashcard Models

struct Flashcard: Identifiable, Codable, Hashable {
    let id: Int
    let question: String
    let answer: String
}

struct FlashcardSet: Identifiable, Codable, Hashable {
    let id: String
    let title: String
    let description: String
    let category: String
    let slug: String
    let language: String?
    let cards: [Flashcard]

    var cardsCount: Int {
        cards.count
    }

    /// Display title for cards - "DMV Flashcards 1", "DMV Flashcards 2", etc.
    var displayTitle: String {
        // Extract number from ID (e.g., "dmv-flashcards-1" -> "1")
        if let lastComponent = id.split(separator: "-").last,
           let number = Int(lastComponent) {
            return "DMV Flashcards \(number)"
        }
        return title
    }
}

File: android-native/app/src/main/java/com/dmvcalifornia/app/core/model/Flashcard.kt
package com.dmvcalifornia.app.core.model

import kotlinx.serialization.Serializable

@Serializable
data class Flashcard(
    val id: Int,
    val question: String,
    val answer: String
)

@Serializable
data class FlashcardSet(
    val id: String,
    val title: String,
    val description: String,
    val category: String,
    val slug: String,
    val language: String? = null,
    val cards: List<Flashcard>
) {
    val cardsCount: Int get() = cards.size

    /** Display title: "DMV Flashcards 1", "DMV Flashcards 2", etc. */
    val displayTitle: String
        get() {
            val lastComponent = id.split("-").lastOrNull()
            val number = lastComponent?.toIntOrNull()
            return if (number != null) "DMV Flashcards $number" else title
        }
}
